tests/kafkatest/tests/core/replication_test.py

```diff
@@ -138,10 +138,11 @@ def min_cluster_size(self):
     @parametrize(failure_mode="hard_bounce",
             broker_type="leader",
             security_protocol="SASL_SSL", client_sasl_mechanism="SCRAM-SHA-256", interbroker_sasl_mechanism="SCRAM-SHA-512")
-    def test_replication_with_broker_failure(self, failure_mode, security_protocol,
-                                             broker_type, client_sasl_mechanism="GSSAPI",
-                                             interbroker_sasl_mechanism="GSSAPI",
-                                             enable_idempotence=False):
+    @matrix(failure_mode=["clean_shutdown", "hard_shutdown", "clean_bounce", "hard_bounce"],
+            security_protocol=["PLAINTEXT"], broker_type=["leader"], compression_type=["gzip"])
```

Comment:
Since we added a line for `enable_idempotence=[True]` recently, I wonder if we can remove `PLAINTEXT` from the following:

```
     @matrix(failure_mode=["clean_shutdown", "hard_shutdown", "clean_bounce", "hard_bounce"],
             broker_type=["leader"],
             security_protocol=["PLAINTEXT", "SASL_SSL"])
```

Or maybe remove that `matrix` line altogether and change `PLAINTEXT` to `SASL_SSL` in the compression path or the idempotent path.

Thoughts?